### PR TITLE
PR: moving db to services

### PR DESCRIPTION
**Pull request by hbertrand**

### CODE REVIEW

File: buster/docparser.py [hunk 1]
```diff
@@ -148,23 +148,24 @@ def generate_embeddings_parser(root_dir: str, output_filepath: str, source: str)
...
+    documents: pd.DataFrame,
+    documents_manager: DocumentsManager,
+):
     logger.info("Preparing database...")
-    documents_manager = get_documents_manager_from_extension(output_filepath)(output_filepath)
     sources = documents["source"].unique()
     for source in sources:
         documents_manager.add(source, documents)
-    logger.info(f"Documents saved to: {output_filepath}")
+    logger.info(f"Documents saved to documents manager")
```
Comment (jerpint) [inline buster/docparser.py:159]: should each documents manager have a `__name__` property so we can differentiate them in the logs?

File: buster/documents/service.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                "title": row["title"],
+                "url": row["url"],
+                "content": row["content"],
+                "n_tokens": row["n_tokens"],
+                "source_id": source_id,
+            }
+            document_id = str(self.db.documents.insert_one(document).inserted_id)
+            self.index.upsert([(document_id, row["embedding"].tolist(), {"source": source})])
+
+    def update_source(self, source: str, display_name: str = None, note: str = None):
```
Comment (jerpint) [inline buster/documents/service.py:51]: we probably need a `delete` or `remove_source` kind of method too

File: buster/docparser.py (hunk 1 above)
Comment (hbertrand) [reply buster/docparser.py:159]: good point, although it's `__repr__` that must be defined for this.

File: buster/documents/service.py (hunk 2 above)
Comment (hbertrand) [reply buster/documents/service.py:51]: we do, for now we can delete from the websites. Will go in another PR

Comment (hbertrand) [reply buster/documents/service.py:58]: In the end I added it to this PR.

**Comment by hbertrand** [comment]
I reverted the change to gradio_app.

File: buster/documents/service.py (hunk 2 above)
Comment (jerpint) [reply buster/documents/service.py:58]: One last thing (maybe for a separate PR) would be add/delete entries from a given source

**Review by jerpint — APPROVED** [review]

File: buster/documents/service.py (hunk 2 above)
Comment (hbertrand) [reply buster/documents/service.py:58]: I'll make a note for another PR

**Pull request merged**